File: app/src/main/java/at/shockbytes/dante/util/tracking/Tracker.java
package at.shockbytes.dante.util.tracking;

import at.shockbytes.dante.util.books.Book;

/**
 * @author Martin Macheiner
 *         Date: 01.06.2017.
 */

public interface Tracker {

    void trackOnScanBook();

    void trackOnScanBookCanceled();

    void trackOnBookManuallyEntered();

    void trackOnFoundBookCanceled();

    void trackOnBookShared();

    void trackOnBackupMade();

    void trackOnBackupRestored();

    void trackOnBookScanned(Book b);

    void trackOnBookMovedToDone(Book b);

}

File: app/src/main/java/at/shockbytes/dante/util/tracking/Tracker.kt
package at.shockbytes.dante.util.tracking

import at.shockbytes.dante.util.tracking.event.DanteTrackingEvent

/**
 * Author:  Martin Macheiner
 * Date:    30.08.2018
 */
interface Tracker {

    fun trackEvent(event: DanteTrackingEvent)
}